packages/shader-lab/src/parser/builtin/functions.ts

```
@@ -20,10 +20,92 @@ export enum EGenType {
+  // Varies along the underlying scalar type: float / int / uint (sometimes with bool)
+  ScalarType
+}
+
+type FamilySpec = {
+  readonly dimension: GenericDimension;
+  readonly members: readonly NonGenericGalaceanType[];
+};
+
+const FamilyMembers: Partial<Record<GenericType, FamilySpec>> = {
+  // Size-varying families: index 0 → scalar, 1 → vec2, 2 → vec3, 3 → vec4
+  [GenericType.GenType]: { dimension: GenericDimension.Size, members: [Keyword.FLOAT, Keyword.VEC2, Keyword.VEC3, Keyword.VEC4] },
```

ESLint

[error] Replace `·dimension:·GenericDimension.Size,·members:·[Keyword.FLOAT,·Keyword.VEC2,·Keyword.VEC3,·Keyword.VEC4]` with `⏎····dimension:·GenericDimension.Size,⏎····members:·[Keyword.FLOAT,·Keyword.VEC2,·Keyword.VEC3,·Keyword.VEC4]⏎·`

(prettier/prettier)